benchmarks/test.nim

```diff
@@ -1,39 +1,22 @@
-import zero_functional, strutils, sequtils, bench, times, os
+import zero_functional, strutils, bench, times, os
 
-var data = readFile("data.txt")
-
-let a = data.split(" ") --> map(parseInt(it)) --> to(seq[int])
-let b = (0..<2_000_000) --> to(seq[int])
-
-proc f(a: int, b: int): int =
-  a + b
+let a = readFile("data.txt").split(" ") --> map(uint8(parseInt(it))) --> to(seq[uint8])
+let b = 0'u32..<2_000_000'u32 --> to(seq[uint32])
 
+proc f(a: uint, b: uint): uint =
+  result = a + b
 
 proc example0: bool =
-  var n = zip(a, b) -->
-                map(f(it[0], it[1])).
-                filter(it mod 4 > 1).
-                map(it * 2).
-                all(it > 4)
-  result = n
-
-proc example1: bool =
-  var o = a -->
+  result = zip(a, b) -->
+              map(f(it[0], it[1])).
+              filter((it mod 4'u) > 1'u).
```

Comment:
for unsigned the compiler is able to convert `(it mod 4'u) > 1'u` to `(it and 3'u) > 1'u` - or maybe even `(it and 2'u) == 2'u`

Reply:
that's a dirty trick for a benchmark!

Reply:
well - but the rust code also contains unsigned operations. So either we use (signed) int for all files or unsigned for all... 😉